openfe/setup/chemicalsystem_generator/easy_chemicalsystem_generator.py

```diff
@@ -112,7 +112,7 @@ def __call__(
                 RFEComponentLabels.PROTEIN: self.protein,
             }
             for i, c in enumerate(self.cofactors):
-                components.update({f'{RFEComponentLabels.COFACTOR}{i+1}': c})
+                components.update({f'{RFEComponentLabels.COFACTOR.value}{i+1}': c})
```

Comment:
what is the reason for using the value here instead of using the `RFEComponentLabels` object itself?

Reply:
For some reason using it like this gives an unexpected label, the cofactor is called `RFEComponentLabels.COFACTOR1` whereas the ligand and protein components are labelled `ligand` and `protein` as expected. See the [error](https://github.com/OpenFreeEnergy/openfe/actions/runs/12300258258/job/34327983233?pr=1048#step:10:2448) for how this looks, I guess its not super important to change as I don't think these labels are used for anything but it doesn't look nice and its not easy to pull out the cofactor with the inconsistent naming.

Reply:
within `abstract_chemicalsystem_generator.py`, if you make the class inherit from `StrEnum` instead of both `str` and `Enum` (i.e.`RFEComponentLabels(StrEnum)`), it'll solve this! 

It looks like `StrEnum` has some additional handling for string representation, but I don't quite understand the details.

Reply:
Ahhh thanks, though I think that the component keys should all be strings going by the [type hint](https://github.com/OpenFreeEnergy/gufe/blob/339143f86ffe234e474ac3313a9fb3ddb90fca98/gufe/chemicalsystem.py#L14) in gufe so maybe they should all be using `.value`?